### PR TITLE
Jkantor/list workflow

### PR DESCRIPTION
**Pull request by jansenk**
**TL;DR -** Add an endpoint for listing submissions for a particular ORA block, for the purpose of the enhanced staff grader

JIRA: [JIRA-XXXX](https://openedx.atlassian.net/browse/JIRA-XXXX)

**What changed?**

- added a new Xblock json handler `list_staff_workflows` to the `staff_grader_mixin`
- The endpoint looks up StaffWorkflows for the given block, and then pulls in additional required data, like grading status, lock status, grade, etc
- example response:
```
{'2a378fb3-f24a-4590-a460-68e183fe3385': {'dateGraded': '2020-03-04 12:35:00+00:00',
                                          'dateSubmitted': '2020-03-02 12:35:00+00:00',
                                          'gradedBy': 'SWLV_staff_1_username',
                                          'gradingStatus': 'graded',
                                          'lockStatus': 'unlocked',
                                          'score': {
                                          	'points_earned': 4,
                                            'points_possible': 6
                                           },
                                          'submission_uuid': '2a378fb3-f24a-4590-a460-68e183fe3385',
                                          'username': 'SWLV_learner_1_username'},
'36fb40ef-074b-4a9c-9e05-8d4095c42e52': {'dateGraded': '2020-03-04 12:35:00+00:00',
                                          'dateSubmitted': '2020-03-02 12:35:00+00:00',
                                          'gradedBy': 'SWLV_staff_0_username',
                                          'gradingStatus': 'graded',
                                          'lockStatus': 'unlocked',
                                          'score': {'points_earned': 6,
                                                    'points_possible': 6},
                                          'submission_uuid': '36fb40ef-074b-4a9c-9e05-8d4095c42e52',
                                          'username': 'SWLV_learner_0_username'},
 '948f99c3-f869-4f17-8521-8965a2f078f3': {'dateGraded': '2020-03-04 12:35:00+00:00',
                                          'dateSubmitted': '2020-03-02 12:35:00+00:00',
                                          'gradedBy': 'SWLV_staff_2_username',
                                          'gradingStatus': 'graded',
                                          'lockStatus': 'unlocked',
                                          'score': {'points_earned': 2,
                                                    'points_possible': 6},
                                          'submission_uuid': '948f99c3-f869-4f17-8521-8965a2f078f3',
                                          'username': 'SWLV_learner_2_username'},
 'cc229fa0-f27e-4b14-9a64-826704011891': {'dateGraded': 'None',
                                          'dateSubmitted': '2020-03-02 12:35:00+00:00',
                                          'gradedBy': '',
                                          'gradingStatus': 'ungraded',
                                          'lockStatus': 'unlocked',
                                          'score': {},
                                          'submission_uuid': 'cc229fa0-f27e-4b14-9a64-826704011891',
                                          'username': 'SWLV_learner_3_username'}}
```
- refactored CSVWriter._map_anonymized_ids_to_usernames out of the class to just exist as its own function
- added a `currently_active` method to SubmissionLock which returns an unevaluated queryset that only includes "currently active" locks

**Developer Checklist**

- [ ] Reviewed the [release process](https://github.com/edx/edx-ora2/blob/master/.github/release_process.md)
- [ ] Translations and JS/SASS compiled
- [ ] Bumped version number in [setup.py](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/setup.py#L39) and [package.json](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/package.json#L3)

**Testing Instructions**
- TBD

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

**Comment by jansenk** [comment]
NOTE: This is going to fail for a bit, because it requires https://github.com/edx/edx-ora2/pull/1691 to function

File: openassessment/xblock/static/dist/manifest.json [hunk 1]
```diff
@@ -4,8 +4,8 @@
   "openassessment-editor-textarea.js.map": "/openassessment-editor-textarea.e855a75d3004c5ff54ee.js.map",
   "openassessment-editor-tinymce.js": "/openassessment-editor-tinymce.f7c8d3bff424e0efaeee.js",
   "openassessment-editor-tinymce.js.map": "/openassessment-editor-tinymce.f7c8d3bff424e0efaeee.js.map",
-  "openassessment-lms.js": "/openassessment-lms.aba4112d7bf75980425a.js",
-  "openassessment-lms.js.map": "/openassessment-lms.aba4112d7bf75980425a.js.map",
+  "openassessment-lms.js": "/openassessment-lms.41ca7aa5d2056aa114a5.js",
```
Comment (leangseu-edx) [inline openassessment/xblock/static/dist/manifest.json:7]: did the `js` files for this got change? if it did, I am not seeing part of the commit.

File: openassessment/staffgrader/tests/test_list_staff_workflows.py [hunk 2]
```diff
@@ -0,0 +1,629 @@
...
+
+
+class TestStaffWorkflowListViewBase(XBlockHandlerTestCase):
+    """ Setup and helper functions for list_staff_workflow tests """
+
+    def setUp(self):
+        super().setUp()
+        # A lot of this test invloves comparing fairly large dicts, so we want to be able to see the whole diff,
+        # no matter what.
+        self.maxDiff = None
```
Comment (nsprenkle) [inline openassessment/staffgrader/tests/test_list_staff_workflows.py:56]: Ooh 😮

File: openassessment/staffgrader/tests/test_list_staff_workflows.py [hunk 3]
```diff
@@ -0,0 +1,629 @@
...
+    # @contextmanager
+    # def _mock_get_team_ids_by_team_submission_uuid(self):
+    #     """
+    #     Context manager that patches get_team_ids_by_team_submission_uuid.
+    #     TODO: It currently has no real return value and is only used to assert that the function was not called.
+    #     """
+    #     with patch(
+    #         'openassessment.staffgrader.staff_grader_mixin.get_team_ids_by_team_submission_uuid',
+    #     ) as patched_map:
+    #         yield patched_map
```
Comment (nsprenkle) [inline openassessment/staffgrader/tests/test_list_staff_workflows.py:141]:
```suggestion
```
